### DISCUSSION TITLE
Insert/modify AST node API

### DISCUSSION THREAD

**Discussion by bombillazo** [graphql/graphql-js#3268] [Q&A] [answered]
Hello, is there a feature in GraphQL js to add a node to an AST document like this one that Ruby [AST Visitor](https://graphql-ruby.org/language_tools/visitor.html#insert-a-node) has?
I can't find it anywhere in the GraphQL js language package, seems like a major feature to miss on handling AST.

**Comment by martinbonnin** [comment, marked as answer]
There is `visit`: https://www.graphql-js.org/api-v16/language/#visit
